homeassistant/components/light/rpi_gpio_pwm.py

```diff
@@ -115,8 +115,12 @@ def __init__(self, led, name):
             self._is_on = (last_state.state == STATE_ON)
             self._brightness = last_state.attributes.get('brightness',
                                                          DEFAULT_BRIGHTNESS)
-            self._led.set(is_on=self._is_on,
-                          brightness=_from_hass_brightness(self._brightness))
+
+            def restore_state():
+                """Set the led into the last known state."""
+                brightness = _from_hass_brightness(self._brightness)
+                self._led.set(is_on=self._is_on, brightness=brightness)
+            await self.hass.async_add_executor_job(restore_state)
```

Comment:
The problem in this case is that the used library resets the led to a default state during initialization. That's why I want to update the physical device with the restored state. 

Is there any other hook or something like this to perform such a operation on start up?

Reply:
Two wrongs don't make a right :)

I think you could use the `HOMEASSISTANT_START` event to make an automation that sets the light state to the existing state (which should be restored at that point).

Reply:
So you propose to add a automation to my hass configuration? I would prefer to a solution that is part of the component itself so that it works for everyone that uses it.

Reply:
The library should handle this with its own persistence logic.

Reply:
I agree with @MartinHjelmare, that is not what should be solved on home assistant.

Reply:
Unfortunately, at least when controlling leds via the PCA9685, which is one of two possibilities provided by this component, this is technically not possible to read the current state of the leds. That's why this cannot be implemented as part of the library.

So when this should not be part of HASS, support for restoring the state needs to be dropped.

Reply:
@soldag Would it be possible to change the library to not reset the lights?

Reply:
@soldag Are you still working on this?

Reply:
@amelchio Sorry, I was kind of busy recently.

The library itself uses https://github.com/adafruit/Adafruit_Python_PCA9685 for controlling LEDs connected to a PCA9685, which does not support reading the current state of the pins, but resets them when initializing the connection. Therefore, it is not easily possible to implement this in the library.

So if you insist, I would need to remove support for restoring the state completely from this component in hass.

Reply:
A reset when Home Assistant is restarted is a bad user experience. It is not clear to me if it is even an option to not do the reset or if it is somehow a part of the protocol?

If the reset can be left out, it seems viable to inline that library in python-pwmled.

Reply:
Yeah that's why I implemented restoration of the state in homeassistant in the first place.

The library doesn't offer the possibility to prevent the reset during initialization.

Reply:
You keep saying "the library" which makes me think that the hardware does support it and the library can be fixed.